### DISCUSSION TITLE
Lnd doesn't recreate admin macaroon unless invoice & readonly macaroons are deleted too

### DISCUSSION THREAD

**Discussion by sangaman** [lightningnetwork/lnd#7566] [Ideas]
Right now, if you delete only the `admin.macaroon` file and restart lnd, the `admin.macaroon` file won't be regenerated. WIthout it, things like `lncli` won't work. If you delete both `readonly.macaroon` and `invoice.macaroon` then lnd will regenerate all three.

I figure lnd should regenerate these independently, or at least the admin macaroon since it's an important part of using lnd (and without it, you can't bake macaroons).

Does this make sense? I wanted to get some input before working on a PR since this isn't really a bug.

**Comment by guggero** [comment]
Yeah, this original default behavior is quite confusing, especially as we added more subserver macaroons which aren't treated the same way as the three "default" macaroons. So I think it makes sense to unify the behavior and re-generate individual macaroons if they are missing.
